apps/judicial-system/web/src/routes/Prison/IndictmentOverview/IndictmentOverview.tsx

```diff
@@ -16,18 +18,40 @@ import {
   RenderFiles,
 } from '@island.is/judicial-system-web/src/components'
 import { CaseFileCategory } from '@island.is/judicial-system-web/src/graphql/schema'
-import { useFileList } from '@island.is/judicial-system-web/src/utils/hooks'
+import {
+  useDefendants,
+  useFileList,
+} from '@island.is/judicial-system-web/src/utils/hooks'
 
 import { strings } from './IndictmentOverview.strings'
 
 const IndictmentOverview = () => {
   const { workingCase } = useContext(FormContext)
   const { formatMessage } = useIntl()
+  const { limitedAccessUpdateDefendant } = useDefendants()
 
   const { onOpen } = useFileList({
     caseId: workingCase.id,
   })
 
+  const { defendants } = workingCase
+  const defendant =
+    defendants && defendants?.length > 0 ? defendants[0] : undefined
+  const [selectedPunishmentType, setPunishmentType] = useState<PunishmentType>()
```

Comment:
Consider updating the workingCase state with the new punishment type choice, that way you don't need to have the punishment type as a state here. This works perfectly well though and within this context you don't need to change it, but the reason we usually update the workingCase state when doing updates is that sometimes we have other things within a component that rely on the workingCase to make decisions, and we've run in to issues where the workingCase state is incorrect because we didn't update it accordingly after a related update and causes weird behaviour.

It would be something like this if you changed it 

```
const IndictmentOverview = () => {
  const { workingCase, setWorkingCase } = useContext(FormContext)
  const { limitedAccessUpdateDefendant, updateDefendantState } = useDefendants()
  
  const { defendants } = workingCase
  const defendant =
    defendants && defendants?.length > 0 ? defendants[0] : undefined

  const onChange = (updatedPunishmentType: PunishmentType) => {
    if (!defendant) return

    const defendantUpdate = {
      defendantId: defendant.id,
      caseId: workingCase.id,
      punishmentType: updatedPunishmentType,
    }
    updateDefendantState(defendantUpdate, setWorkingCase)
    limitedAccessUpdateDefendant(defendantUpdate)
  }
```

and then you could also replace the checked logic with 

`        checked={
                  defendant?.punishmentType ===
                  PunishmentType.INDICTMENT_RULING_DECISION_FINE
                }`

if you wanted to remove the hasSetPunishmentType function

Reply:
This is just a suggestion though so it's up to you which way you go, both work 😊 Just wanted to point it out because it's a pattern you'll notice a lot that the workingCase gets updated either directly before or after the update, or as part of a "setAndSendToServer" hook which performs both at once

Reply:
Aha makes sense, I will address this since I also like to simplify the `boolean` checked property 👌